src/logic/calculate.test.js

```diff
@@ -0,0 +1,30 @@
+import calculate from './calculate';
+
+describe('calculate function', () => {
+  it('should handle AC button', () => {
+    const result = calculate({ total: '10', next: '5', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+
+  it('should handle number buttons', () => {
```

Comment:
- [ ] I can see some of the functionalities & scenarios of the operate.js and calculate.js modules is being covered using Jest but you might have forgotten that you need to also need to make use of Jest snapshots for testing all the components, none of the components is tested. Remember to also make use of React Testing Library to simulate user interaction.